src/mem/Pool.h

```diff
@@ -190,7 +186,7 @@ class MemAllocator
      \param stats   Object to be filled with statistical data about pool.
      \retval        Number of objects in use, ie. allocated.
      */
-    virtual int getStats(MemPoolStats * stats, int accumulate = 0) = 0;
+    virtual int getStats(MemPoolStats * stats) = 0;
```

Comment:
Since we are going to change every method, please also fix the type of the first parameter. If it is not going to trigger lots of changes, please fix the return type as well:

```suggestion
    virtual size_t getStats(MemPoolStats &stats) = 0;
```

If you want to avoid adding many s/pointer/reference/ changes to the method implementations while this PR is getting reviewed, add temporary diff reducers to those implementations. For example:

```C++
    const auto stats = &statsReference; // TODO: Remove this diff reducer before merging
```

Reply:
Done in [fdf49db](https://github.com/squid-cache/squid/pull/1186/commits/fdf49db060d1b6529ac0a48a4925768662c339d3) and [1d1b2aa](https://github.com/squid-cache/squid/pull/1186/commits/1d1b2aa7f65c0bf400f4c44278b23b31bbcb0ac1)